packages/svelte-vscode/syntaxes/svelte.tmLanguage.src.yaml

```diff
@@ -509,16 +509,20 @@ repository:
   # Scopes the `name` part in `<name>`.
   tags-name:
     patterns:
-    # Svelte (`svelte:<type>`) elements.
-    - match: '(svelte)(:)([a-z][\w0-9:-]*)'
+    # Svelte built-in elements (e.g., svelte:self, svelte:component).
+    - match: '(svelte)(:)([a-z][\w:-]*)'
       captures:
         1: { name: keyword.control.svelte }
         2: { name: punctuation.definition.keyword.svelte }
         3: { name: entity.name.tag.svelte }
     # Slot.
     - { match: 'slot', name: keyword.control.svelte }
-    # Components.
-    - { match: '[A-Z][a-zA-Z0-9_]*', name: support.class.component.svelte }
+    # Components (either Namespaced.Component, namespaced.component or PascalCase).
+    - match: '([A-Z][\w]+)|(?<=\.)([\w]+)(?<!\.)|(?!\.)([\w]+)(?=\.)'
+      captures:
+        1: { name: support.class.component.svelte }
+        2: { name: support.class.component.svelte }
+        3: { name: support.class.component.svelte }
```

Comment:
how about creating a regex that includes the dot (that regex will be simpler/faster) and then use a simple `patterns` sub expression to highlight the dots differently?

Reply:
That was my first attempt, but I thought we needed to have `.` not highlighted. I will try with your suggestion!

Reply:
I used a simpler regex and used a sub pattern match [09631](https://github.com/sveltejs/language-tools/pull/2552/commits/09631a433dbeaf2f75386442d808ce07526af096)